modin/pandas/test/utils.py

```diff
@@ -868,12 +868,43 @@ def generate_none_dfs():
     return df, df2
 
 
+def make_dict_hash(dict_to_hash):
+    """Makes a hash from a dictionary considering nested
+    dictionaries, lists, sets or callables.
+
+    Parameters
+    ----------
+    dict_to_hash: dict
+        Dictionary that should be hashed.
+
+    Returns
+    -------
+        Dictionary hash.
+    """
+
+    def _make_hash(obj):
+        if isinstance(obj, (set, list, dict)):
+            object_hash = hash(frozenset(obj))
+        elif callable(obj):
+            object_hash = hash(obj.__name__)
+        else:
+            # conversion to str is done because hash(5) == hash(5.0)
+            object_hash = hash(obj if isinstance(obj, str) else str(obj))
+
+        return object_hash
+
+    # process unhashed values from dict_to_hash
+    new_dict = {key: _make_hash(value) for key, value in dict_to_hash.items()}
+    return hash(frozenset(new_dict))
+
+
 def get_unique_filename(
     test_name: str,
     kwargs: dict = {},
     extension: str = "csv",
     data_dir: str = IO_OPS_DATA_DIR,
     suffix: str = "",
+    debug_mode=False,
```

Comment:
How should a developer use this option?
Please describe the scenario.

Reply:
For example after test are finished, some of test files may not be deleted (for some unknown reason) and with only hash name it will be difficult to find the test that generated this file.

Reply:
Previously that function always worked in `debug_mode=True`, why do we need an ability to disable it?

Reply:
`get_unique_filename` in `debug_mode` is based on manual checking of `bad` kwargs values (`lambda` function for example) and replacing with some `str`. This works fine, but potentially it can cause unexpected behavior or bugs. In the case if we use `regular` mode we always obtain predictable result with fixed length.